### PR TITLE
feat: order again

### PR DESCRIPTION
**Pull request by FabianGerke**
## Why:

Closes: [AB#114956](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/114956)

## Checklist before requesting a review

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/components/OrderAgain/OrderAgain.vue [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      type: 'positive',
+      message: t('addedToCart'),
+    });
+    goToPage(paths.cart);
+  }
+  loadingAddToCart.value = false;
+};
+</script>
+
+<style scoped>
```
Comment (csandru-plenty) [inline apps/web/components/OrderAgain/OrderAgain.vue]: Do you think these are reusable? Does it make sense moving into a class here:
```apps/web/assets/style.scss```

File: apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <div role="status" class="flex w-full mt-6 items-center animate-pulse">
+    <div
+      class="flex items-center justify-center w-24 h-24 flex-shrink-0 bg-gray-300 text-gray-600 rounded-sm mr-3 sm:w-40 sm:h-40"
+    >
+      <svg
```
Comment (csandru-plenty) [inline apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue]: We should probably stick to our [default way](https://github.com/plentymarkets/plentyshop-pwa-docs/blob/main/guide/how-to/custom-icon.md)

File: apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      </svg>
+    </div>
+    <div class="w-full flex flex-col">
+      <div class="h-6 sm:h-8 bg-gray-300 rounded-sm w-full"></div>
+      <div class="h-6 sm:h-8 bg-gray-300 rounded-sm w-[140px] mt-2"></div>
+      <div class="h-4 bg-gray-300 rounded-sm w-11/12 mt-2"></div>
+      <div class="h-4 bg-gray-300 rounded-sm w-7/12 mt-1"></div>
+      <div class="h-6 sm:h-8 bg-gray-300 rounded-sm w-[150px] mt-2"></div>
+    </div>
+    <span class="sr-only">Loading...</span>
```
Comment (csandru-plenty) [inline apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue]: Translation?

File: apps/web/composables/useOrderAgain/types.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import type { Order } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/composables/useOrderAgain/types.ts]: I would double-check & refactor all these ```type``` imports, as some of them are not accurate, leading to unexpected results (eg: here we are dealing with an interface)

```import { type Xxxxx, Yyyy };``` is more appropiate imo

File: apps/web/composables/useOrderAgain/useOrderAgain.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
+import type { BasketItemOrderParamsProperty, DoAddItemParams, Order } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/composables/useOrderAgain/useOrderAgain.ts]: I would double-check & refactor all these ```type``` imports, as some of them are not accurate, leading to unexpected results

```import { type Xxxxx, Yyyy };``` is more appropiate imo

File: apps/web/components/OrderAgain/OrderAgain.vue (hunk 1 above)
Comment (FabianGerke) [reply apps/web/components/OrderAgain/OrderAgain.vue]: Could be something we can use more often in the future, I will move it.

File: apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue (hunk 3 above)
Comment (FabianGerke) [reply apps/web/components/Skeletons/OrderAgainItem/OrderAgainItem.vue]: I removed that part, we dont need it actually.

File: apps/web/components/OrderAgain/types.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import type { Order } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/components/OrderAgain/types.ts]: here import is an interface

File: apps/web/components/OrderAgainButton/types.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import type { Order } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/components/OrderAgainButton/types.ts]: here import is an interface

File: apps/web/composables/useCart/useCart.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import type { CartItem } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/composables/useCart/useCart.ts]: here on all imports we have only a few ```type```, rest of them interfaces

File: apps/web/composables/useOrderAgain/useOrderAgain.ts [hunk 9]
```diff
@@ -0,0 +1,118 @@
+import { BasketItemOrderParamsProperty, DoAddItemParams, Order, orderGetters } from '@plentymarkets/shop-api';
```
Comment (csandru-plenty) [inline apps/web/composables/useOrderAgain/useOrderAgain.ts]: here on all imports we have only a few ```type```, rest of them interfaces

File: apps/web/components/OrderAgain/OrderAgain.vue
Comment (csandru-plenty) [inline apps/web/components/OrderAgain/OrderAgain.vue:1]: Firefox & Chrome
Screenshot 2024-07-17 at 08 41 52
Screenshot 2024-07-17 at 09 07 39

File: apps/web/composables/useCart/useCart.ts (hunk 8 above)
Comment (FabianGerke) [reply apps/web/composables/useCart/useCart.ts]: As you see I dont changed that in this PR.

Comment (csandru-plenty) [reply apps/web/composables/useCart/useCart.ts]: true, true

**Review by csandru-plenty — DISMISSED** [review]

**Review by maxiroellplenty — APPROVED** [review]

**Pull request merged**